src/api/bundler-namespace.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  UserOperation,
+  UserOperationEstiamateGasResponse,
+  UserOperationReceipt,
+  UserOperationRequest
+} from '../types/types';
+import { AlchemyConfig } from './alchemy-config';
+
+/**
+ * The Bundler namespace contains methods used for interacting performing user operations and
+ * checking on the state of submitted operations.
+ *
+ * Do not call this constructor directly. Instead, instantiate an Alchemy object
+ * with `const alchemy = new Alchemy(config)` and then access the transact
+ * namespace via `alchemy.paymaster`.
+ */
+export class BundlerNamespace {
+  /** @internal */
+  constructor(private readonly config: AlchemyConfig) {}
+
+  /**
+   * Sends a User Operation
+   *
+   * Sends a user operation to the provided bundler client evm network, and
+   * returns a user operation hash on successful submission.
+   *
+   * @param userOperation - The user operation.
+   * @public
+   */
+  async sendUserOperation(
+    userOperation: UserOperationRequest
+  ): Promise<string> {
+    const provider = await this.config.getProvider();
+    return provider._send(
+      'eth_sendUserOperation',
+      [userOperation],
+      'sendUserOperation'
+    );
+  }
+
+  /**
+   * Get a User Operation by Hash
+   *
+   * This method returns the user operation object based on the
+   * provided user operation hash.
+   *
+   * @param userOpHash - The hash of the user operation.
+   * @public
+   */
+  async getUserOperationByHash(userOpHash: string): Promise<UserOperation> {
+    const provider = await this.config.getProvider();
+    return provider._send(
+      'eth_getUserOperationByHash',
+      [userOpHash],
+      'getUserOperationByHash'
+    );
+  }
+
+  /**
+   * Get Client Supported Entry Points
+   *
+   * Returns an array of the entry point addresses that are
+   * supported by the Alchemy bundler client.
+   *
+   * @public
+   */
+  async supportedEntryPoints(): Promise<string[]> {
+    const provider = await this.config.getProvider();
+    return provider._send(
+      'eth_supportedEntryPoints',
+      [],
+      'supportedEntryPoints'
+    );
+  }
+
+  /**
+   * Get a User Operation Receipt
+   *
+   * This method returns the user operation receipt based on the
+   * provided user operation hash.
+   *
+   * @param userOpHash - The hash of the user operation.
+   * @public
+   */
+  async getUserOperationReceipt(
+    userOpHash: string
+  ): Promise<UserOperationReceipt> {
+    const provider = await this.config.getProvider();
+    return provider._send(
+      'eth_getUserOperationReceipt',
+      [userOpHash],
+      'getUserOperationReceipt'
+    );
+  }
+
+  /**
+   * Estimates the Gas Values for a UserOperation.
+   *
+   * This method returns gas estimates based on the provided user operation request.
+   *
+   * @param userOperation - The user operation.
+   * @public
+   */
+  async estimateUserOperationGas(
+    userOperation: UserOperationRequest
```

Comment:
yea something like this: https://github.com/OMGWINNING/wallet-services-sdk/pull/1/files#diff-f69e580bf3fc644bf19e546a0a2ae0cd8488e526fbe4e7b24a6b7a81b8df2a7eR69-R71

won't need the resolve properties bit since the type here isn't something with `Promise` fields. But the deeplyHexlify bit is needed: https://github.com/OMGWINNING/wallet-services-sdk/blob/36af2466dbe2afdfbee8cc8e60acfa348e379064/packages/uo-builder/src/deeplyHexlify.ts